core/src/test/kotlin/com/strumenta/kolasu/transformation/ASTTransformerTest.kt

```diff
@@ -116,4 +137,110 @@ class ASTTransformerTest {
             )!!
         )
     }
+
+    /**
+     * Example of transformation to perform a simple type calculation.
+     */
+    @Test
+    fun computeTypes() {
+        val myTransformer = ASTTransformer(allowGenericNode = false).apply {
```

Comment:
Would it make sense to later introduce a specific subtype of ASTTransformer specifically for type calculations?

Reply:
It could make sense, but I think we could also investigate if having type computation and symbol resolution implemented as lazy properties of the AST nodes would come along better.